src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsService.java

```diff
@@ -0,0 +1,610 @@
+    }
+
+    @Override
+    public void configureBuildPlan(ProgrammingExerciseParticipation participation) {
+        final var projectKey = participation.getProgrammingExercise().getProjectKey();
+        final var planKey = participation.getBuildPlanId();
+        updatePlanRepository(projectKey, planKey, null /* not important */, null /* not important */, participation.getRepositoryUrl(), Optional.empty());
+        enablePlan(projectKey, planKey);
+    }
+
+    // TODO this was a bad design choice. We should only have one configureBuildPlan method i.m.o
+    @Override
+    public void updatePlanRepository(String projectKey, String planName, String repoNameInCI, String vcsProject, String vcsRepositoryUrl, Optional<List<String>> triggeredBy) {
+        // remove potential username from repo URL. Jenkins uses the Artemis Admin user and will fail if other usernames are in the URL
+        final var repoUrl = vcsRepositoryUrl.replaceAll("(https?://)(.*@)(.*)", "$1$3");
+        final var config = jobXml(projectKey, planName);
+        final var urlElements = config.getElementsByTagName("url");
+        if (urlElements.getLength() != 2) {
+            throw new IllegalArgumentException("Configuration of build plans currently only supports a model with two repositories, ASSIGNMENT and TESTS");
+        }
+        urlElements.item(1).getFirstChild().setNodeValue(repoUrl);
+
+        final var errorMessage = "Error trying to configure build plan in Jenkins " + planName;
+        postXml(config, String.class, HttpStatus.OK, errorMessage, Endpoint.PLAN_CONFIG, projectKey, planName);
+    }
+
+    @Override
+    public void triggerBuild(ProgrammingExerciseParticipation participation) {
+        final var projectKey = participation.getProgrammingExercise().getProjectKey();
+        final var planKey = participation.getBuildPlanId();
+
+        try {
+            job(projectKey, planKey).build(true);
+        }
+        catch (IOException e) {
+            throw new JenkinsException("Error triggering build: " + planKey, e);
+        }
+    }
+
+    @Override
+    public void deleteProject(String projectKey) {
+        try {
+            jenkinsServer.deleteJob(projectKey, true);
+        }
+        catch (IOException e) {
+            throw new JenkinsException("Error while trying to delete folder in Jenkins for " + projectKey, e);
+        }
+    }
+
+    @Override
+    public void deleteBuildPlan(String projectKey, String buildPlanId) {
+        try {
+            jenkinsServer.deleteJob(folder(projectKey), buildPlanId, true);
+        }
+        catch (IOException e) {
+            throw new JenkinsException("Error while trying to delete job in Jenkins: " + buildPlanId, e);
+        }
+    }
+
+    @Override
+    public String getPlanKey(Object requestBody) throws Exception {
+        final var result = TestResults.convert(requestBody);
+        final var nameParams = result.getFullName().split(" ");
+        if (nameParams.length != 4) {
```

Comment:
Can you give an example of a valid `nameParams`?

Reply:
True, this is pretty confusing otherwise. I added some inline documentation with an example